code/modules/mob/mob_grab_specials.dm

```diff
@@ -147,3 +147,46 @@
 
 	if(can_eat)
 		H.attempt_devour(target, H.eat_types, H.mouth_size)
+
+
+/obj/item/weapon/grab/proc/hair_pull(mob/living/carbon/human/target, mob/attacker, var/target_zone)
+
+
+	var/mob/living/carbon/human/H = target
+	var/obj/item/weapon/grab/G = attacker.get_active_hand()

```

Comment:
Why do you need to create a reference to the grab? Isn't this proc literally part of the grab object? Can't you just past a reference to `src`?